File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev1=1;
        int prev2=1;
        int curr_i=1;
        for (int i =2; i<=n;i++)
        {
            curr_i=prev1+prev2;
            prev1=prev2;
            prev2=curr_i;
        }
        return curr_i;
        
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
   
    def climbStairs(self, n: int) -> int:
#         if n ==0:
#             return 1
#         if n ==1:
#             return 1

#         return self.climbStairs(n-1)+self.climbStairs(n-2)
        dp=[0]*(n+1)
        dp[1]=1
        dp[0]=1
        for i in range(2,n+1):
            dp[i]=dp[i-1]+dp[i-2]
        return dp[n]
        
        
        
    
    
        
        
        